packages/language-server/src/__test__/completion.test.ts

```diff
@@ -3442,6 +3484,130 @@ suite('Completions', function () {
         })
       })
     })
+
+    suite('self-filter', () => {
+      test('Baseline', () => {
+        assertCompletion({
+          schema: /* prisma */ `
+              generator client {
+                provider = "prisma-client-js"
+              }
+              
+              datasource db {
+                provider = "postgresql"
+                url      = env("DATABASE_URL")
+              }
+              
+              model A {
+                id  Int @id @unique @default(autoincrement()) @map("hi") @ignore @db.Integer |
+              }
+              `,
+          expected: {
+            isIncomplete: false,
+            items: [fieldAttributeRelation],
+          },
+        })
+      })
+
+      test('@relation', () => {
+        assertCompletion({
+          schema: /* prisma */ `
+              generator client {
+                provider = "prisma-client-js"
+              }
+              
+              datasource db {
+                provider = "postgresql"
+                url      = env("DATABASE_URL")
+              }
+              
+              model A {
+                id  Int @id
+                b   B   @relation(fields: [bId], references: [id]) |
+                bId Int
+              }
+              
+              model B {
+                id Int @id
+                A  A[]
+              }
+              `,
+          expected: {
+            isIncomplete: false,
+            items: [
+              fieldAttributeDatasourceName,
+              fieldAttributeUnique,
+              fieldAttributeMap,
+              fieldAttributeDefault,
+              fieldAttributeIgnore,
+            ],
+          },
+        })
+      })
+      test('@@ignore filters @ignore', () => {
+        assertCompletion({
+          schema: /* prisma */ `
+            generator client {
+              provider = "prisma-client-js"
+            }
+            
+            datasource db {
+              provider = "postgresql"
+              url      = env("DATABASE_URL")
+            }
+            
+            model A {
+              id Int @default(autoincrement()) @map("hi")
+              name String |
+              
+              @@ignore
+            }
+            `,
+          expected: {
+            isIncomplete: false,
+            items: [
+              fieldAttributeDatasourceName,
+              fieldAttributeId,
+              fieldAttributeUnique,
+              fieldAttributeMap,
+              fieldAttributeDefault,
```

Comment:
Why no `@ignore` here?

Reply:
`@@ignore` supercedes it
<img width="1000" alt="image" src="https://github.com/prisma/language-tools/assets/29753584/5ac02b3b-4f2e-45a1-8afc-d980f1df6730">

Reply:
Oh interesting, I thought this would just not do anything.
All the `@ignore` becoming invalid when you add a `@@ignore` is not the best experience.
But ok 🤷

Reply:
Yeah, I'd agree. Seems to have been like this for a few years at least tho.
Could probably just be a warning that says, hey this does nothing -- or even in an ideal world if we had a config that let users define if _they_ want it to be an error or not